### PR TITLE
ICMSLST-2296 Generate CFS Cover Letter PDF

### PR DESCRIPTION
**Pull request by mikeylockwood**
* CFS cover letter generated with management command `./manage.py generate_static_pdf`
* Link for the cfs cover letter will only be displayed after a cover letter has been generated

### CODE REVIEW

File: config/settings/base.py [hunk 1]
```diff
@@ -227,6 +229,10 @@
     "django_chunk_upload_handlers.s3.S3FileUploadHandler",
 )
 
+# Used for static documents stored in S3
+STATIC_FILES_S3_PREFIX = "static_documents"
+CFS_COVER_LETTER_FILENAME = "CFS Letter"
```
Comment (MattHolmes123) [inline config/settings/base.py:234]: Will these ever need to be used elsewhere / change value?
Could they just be hard coded in `cfs_cover_letter_key_filename` for now?

File: web/management/commands/generate_static_pdf.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            anon_user = get_anonymous_user()
+
+            File.objects.create(
+                is_active=True,
+                filename=filename,
+                content_type="application/pdf",
+                file_size=file_size,
+                path=key,
+                created_by=anon_user,
+            )
```
[the remark this answers is not in the document]
Comment (mikeylockwood) [reply web/management/commands/generate_static_pdf.py:48]: Would be difficult as file_size is required an very unlikely to match for each file

[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/management/commands/generate_static_pdf.py:48]: no thats why you use default

[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/management/commands/generate_static_pdf.py:48]:
```
File.objects.get_or_create(
    filename=filename, path=key,
    defaults={
        "is_active": True, 
        "content_type": "application/pdf",
        "file_size": file_size,
        "created_by": anon_user
    }
)```

[the remark this answers is not in the document]
Comment (mikeylockwood) [reply web/management/commands/generate_static_pdf.py:48]: I didn't know about defaults, that's good to know 👍

File: web/management/commands/generate_static_pdf.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        if not os.path.exists("output_documents"):
+            self.stdout.write("Folder output_documents does not exist. Creating.")
+            os.makedirs("output_documents")
+
+        self.stdout.write("Saving the pdf to output_documents.")
+
+        with open(f"output_documents/{filename}", "wb") as f:
+            f.write(file_obj.getvalue())
+            self.stdout.write(f"Saved the pdf to output_documents/{filename}")
```
Comment (MattHolmes123) [inline web/management/commands/generate_static_pdf.py:62]: Pathlib can be used to make it easier to work with files and directories:
```python
from pathlib import Path
    
output_documents = Path(".") / "output_documents/"
output_documents.mkdir(parents=True, exist_ok=True)
output_file = output_documents / "some_file"

with output_file.open("w") as f:
    f.write('some text')
```

**Review by MattHolmes123 — APPROVED** [review]

**Pull request merged**